modulemd/modulemd-defaults-v1.c

```diff
@@ -1213,60 +1215,69 @@ modulemd_defaults_v1_merge (const gchar *module_name,
-  else if (from->default_stream && !into->default_stream)
+  if (from->default_stream && !into->default_stream)
     {
       modulemd_defaults_v1_set_default_stream (
         merged, from->default_stream, NULL);
     }
-  else if (into->default_stream && from->default_stream)
+  else if (merged->default_stream && from->default_stream)
     {
-      if (g_str_equal (into->default_stream, DEFAULT_MERGE_CONFLICT))
+      if (g_str_equal (merged->default_stream, DEFAULT_MERGE_CONFLICT))
         {
           /* A previous pass over this same module encountered a merge
-           * conflict, so we need to propagate that.
+           * conflict, so keep it.
            */
-          modulemd_defaults_v1_set_default_stream (
-            merged, DEFAULT_MERGE_CONFLICT, NULL);
         }
-      else if (!g_str_equal (into->default_stream, from->default_stream))
+      else if (!g_str_equal (merged->default_stream, from->default_stream))
         {
-          /* They have conflicting default streams */
-          g_info ("Module stream mismatch in merge: %s != %s",
-                  into->default_stream,
-                  from->default_stream);
-          if (strict_default_streams)
+          if (from_modified > into_modified)
             {
-              g_set_error (error,
-                           MODULEMD_ERROR,
-                           MODULEMD_ERROR_VALIDATE,
-                           "Default stream mismatch in module %s: %s != %s",
-                           module_name,
-                           into->default_stream,
-                           from->default_stream);
-              return NULL;
+              modulemd_defaults_v1_set_default_stream (
+                merged, from->default_stream, NULL);
+            }
+          else if (from_modified == into_modified)
+            {
+              /* They have conflicting default streams */
+              g_info ("Module stream mismatch in merge: %s != %s",
+                      into->default_stream,
+                      from->default_stream);
+              if (strict_default_streams)
+                {
+                  g_set_error (
+                    error,
+                    MODULEMD_ERROR,
+                    MODULEMD_ERROR_VALIDATE,
+                    "Default stream mismatch in module %s: %s != %s",
+                    module_name,
+                    into->default_stream,
+                    from->default_stream);
+                  return NULL;
+                }
+              modulemd_defaults_v1_set_default_stream (
+                merged, DEFAULT_MERGE_CONFLICT, NULL);
             }
-          modulemd_defaults_v1_set_default_stream (
-            merged, DEFAULT_MERGE_CONFLICT, NULL);
         }
       else
         {
-          /* They're the same, so store that */
-          modulemd_defaults_v1_set_default_stream (
-            merged, into->default_stream, NULL);
+          /* They're the same, so change nothing */
         }
     }
   else
```

Comment:
The comment below should be changed. If it's reached, it only means that the "from" default stream value is NULL, so there is nothing to do.